docs/building-rpm-ostree-and-autosd.md

```diff
@@ -0,0 +1,171 @@
+<!-- markdownlint-disable-file MD013 MD033 -->
+# Building rpm-ostree in autosd image from scratch
+
+The AutoSD is an upstream repository for Red Hat In-Vehicle OS, much like CentOS Stream is to RHEL. AutoSD is based on CentOS Stream with a few divergences. If your base system is not a CentOS, you can use a container to generate a compatible CentOS Stream 9 RPM as showed below, otherwise just skip the step and use your raw system.
+
+## Building rpm-ostree rpm on CentOS Stream9
+
+Create a **ContainerFile** with the content below.
+
+```bash
+FROM centos:stream9
+
+RUN dnf update -y
+RUN dnf install -y 'dnf-command(config-manager)' epel-release
+RUN dnf config-manager --set-enabled crb
+RUN dnf config-manager --add-repo https://buildlogs.centos.org/9-stream/automotive/aarch64/packages-main/
+RUN dnf config-manager --add-repo https://buildlogs.centos.org/9-stream/autosd/aarch64/packages-main/
+RUN if [ $(uname -m) == "aarch64" ]; then dnf install -y --nogpgcheck autosig-u-boot; fi
+RUN dnf install -y --allowerasing --nogpgcheck mock cpio qemu-kvm rpm-build autoconf \
+  automake bison dracut git gpgme-devel gtk-doc libattr-devel libcap-devel libtool \
+  make openssl-devel fuse libarchive libselinux libsoup zlib \
+  gobject-introspection-devel e2fsprogs-devel fuse-devel libarchive-devel \
+  libcurl-devel libsoup-devel systemd-devel xz-devel flex python3-devel swig bc \
+  centos-release-automotive audit-libs-devel bzip2-devel cryptsetup-devel dbus-devel \
+  elfutils-devel firewalld-filesystem gcc-c++ gettext gnu-efi gnu-efi-devel \
+  gnutls-devel gperf hostname kmod-devel libacl-devel libfdisk-devel libgcrypt-devel \
+  libidn2-devel libmicrohttpd-devel libseccomp-devel libxkbcommon-devel \
+  libzstd-devel lz4 lz4-devel pam-devel perl tree valgrind-devel meson \
+  bash-completion tpm2-tss-devel attr dtc
+
+# Install deps required for recent rpm-ostree
+RUN cd rpmbuild/SPECS/ && curl -OL https://gitlab.com/redhat/centos-stream/rpms/rpm-ostree/-/raw/c9s/rpm-ostree.spec
+RUN cd /tmp && dnf builddep -y rpm-ostree.spec
+
+# Lets use the latest from community (see next curl rustup step).
+# Also avoids "cargo metadata: error: failed to run rustc to learn about target-specific information"
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+```
+
+## Building the image
+
+```bash
+podman build -t rpmostreedevel -f ContainerFile
+```
+
+## Running the image
+
+The strategy now is run the container sharing the current **HOME/rpmbuild** dir, so as soon the rpm is created in the container it will be located in the host machine in the running user dir.
+
+```bash
+sudo podman run --rm --privileged -v /home/$USER/rpmbuild/:/home/$USER/rpmbuild/ -ti rpmostreedevel /bin/bash
+
+[root@d1ec4ed543ec /]#
+```
+
+## Building a forked rpm-ostree
+
+Now time to clone your [rpm-ostree](https://github.com/coreos/rpm-ostree) fork and build it.
```

Comment:
Yeah this is where I find the toolbox approach superior because if you're using `--rm` with your dev container then your git repository is going to get deleted...

Reply:
To be honest, I could go with any approach info if it was easy to find and use. In this case specific, we want to simply change few lines of rpm-ostree generate the rpm and test against autosd. 

I would love to simple do `make rpm` in the branch I am working and generates whatever is needed to create the rpms and share the location to developers to grab the rpms (the tool should be smart enough that doesn't matter if it's in the cloud using github/gitlab  or local box) and after this task is done optionally remove any pod/build/virt used.